src/Evolution/Systems/ScalarWave/TimeDerivative.cpp

```diff
@@ -17,12 +17,16 @@ void TimeDerivative<Dim>::apply(
     const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*> dt_phi,
     const gsl::not_null<Scalar<DataVector>*> dt_psi,
 
+    const gsl::not_null<Scalar<DataVector>*> result_gamma2,
```

Comment:
For my education, can you explain this change (i.e., the first commit of this PR) in a github comment?

The caller already knows gamma2, or it wouldn't be able to pass it into the function. What does returning gamma2 again by not-null accomplish?

Reply:
In order to get the constraint damping parameter on the boundary we need it to be a temporary tag. I'm trying to avoid allowing people to slice everything in the DataBox to the boundary because all you need are the principal part terms of the evolution equations and the evolved (maybe prims too) variables. Ideally I'd like the time derivatives to take the gamma2 function as an argument as well as the coordinates, and then compute it internally. The downside of course is that it would get re-computed every time step, even when that's not needed.

Reply:
Okay, that fills in the gap I was missing -- thank you!

Reply:
Maybe I should rename the parameter for now? Something like `gamma2_for_projection_to_interface`? Lengthy, but documents why it exists :P

Reply:
That is pretty long... but it only appears in a few places, so it wouldn't be too bad. Alternatively, a short comment (maybe not even a doxygen comment, but a simple `//` comment around line 28 below? Up to you, I think either would be helpful :)